hazelcast/src/main/java/com/hazelcast/query/impl/predicates/PartitionPredicateImpl.java

```diff
@@ -124,15 +162,15 @@ public boolean equals(Object o) {
 
         PartitionPredicateImpl<?, ?> that = (PartitionPredicateImpl<?, ?>) o;
 
-        if (partitionKey != null ? !partitionKey.equals(that.partitionKey) : that.partitionKey != null) {
+        if (partitionKeys != null ? !partitionKeys.equals(that.partitionKeys) : that.partitionKeys != null) {
```

Comment:
Can be simplified like this: ` if (!Objects.equals(partitionKeys, that.partitionKeys)) {`?